Commands/Files/AddFile.cs

```diff
@@ -124,6 +132,47 @@ protected override void ExecuteCmdlet()
 
                 ClientContext.ExecuteQueryRetry();
             }
+            if (ContentType != null)
+            {
+                ClientContext.Load(file, f=>f.ListId);
+                ClientContext.ExecuteQueryRetry();
+
+                List list = SelectedWeb.Lists.GetById(file.ListId);
+                ClientContext.Load(list);
+                ClientContext.ExecuteQueryRetry();
+
+                ContentType ct;
+                if (!string.IsNullOrEmpty(ContentType.Id))
+                {
+                    ct = list.GetContentTypeById(ContentType.Id);
+                }
+                else if(!string.IsNullOrEmpty(ContentType.Name))
+                {
+                    ct = list.GetContentTypeByName(ContentType.Name);
+                }
+                else if(ContentType.ContentType != null)
+                {
+                    ct = ContentType.ContentType;
+                }
+                else
+                {
+                    ct = null;
+                   
+                }
+
+                if (ct != null)
+                {
+                    var item = file.ListItemAllFields;
+                    item["ContentTypeId"] = ct.Id.StringValue;
+                    item.Update();
+                    ClientContext.ExecuteQueryRetry();
+                }
+                else
+                {
+                    WriteVerbose("Content Type was 'null' was not found.");
```

Comment:
I am going to go ahead and throw an invalid argument exception if a content type is specified but is not available on the list. This will keep invalid data from being posted, and act to notify the user that the content type they want isn't where they think it is.